vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -635,47 +657,51 @@ class OnboardingViewModel @AssistedInject constructor(
-                            updateSignMode(SignMode.SignIn)
-                            when (vectorFeatures.isOnboardingCombinedLoginEnabled()) {
-                                true  -> _viewEvents.post(OnboardingViewEvents.OpenCombinedLogin)
-                                false -> _viewEvents.post(OnboardingViewEvents.OnSignModeSelected(SignMode.SignIn))
-                            }
-                        }
-                        OnboardingFlow.SignUp -> {
-                            updateSignMode(SignMode.SignUp)
-                            internalRegisterAction(RegisterAction.StartRegistration, ::emitFlowResultViewEvent)
-                        }
-                        OnboardingFlow.SignInSignUp,
-                        null                  -> {
-                            _viewEvents.post(OnboardingViewEvents.OnLoginFlowRetrieved)
-                        }
+        }
+    }
+
+    private suspend fun onHomeServerSelected(config: HomeServerConnectionConfig, serverTypeOverride: ServerType?, authResult: StartAuthenticationResult) {
+        updateServerSelection(config, serverTypeOverride, authResult)
+        if (authResult.selectedHomeserver.preferredLoginMode.supportsSignModeScreen()) {
+            when (awaitState().onboardingFlow) {
+                OnboardingFlow.SignIn -> {
+                    updateSignMode(SignMode.SignIn)
+                    when (vectorFeatures.isOnboardingCombinedLoginEnabled()) {
+                        true  -> _viewEvents.post(OnboardingViewEvents.OpenCombinedLogin)
+                        false -> _viewEvents.post(OnboardingViewEvents.OnSignModeSelected(SignMode.SignIn))
                     }
-                } else {
+                }
+                OnboardingFlow.SignUp -> {
+                    updateSignMode(SignMode.SignUp)
+                    internalRegisterAction(RegisterAction.StartRegistration, ::emitFlowResultViewEvent)
+                }
+                OnboardingFlow.SignInSignUp,
+                null                  -> {
                     _viewEvents.post(OnboardingViewEvents.OnLoginFlowRetrieved)
                 }
             }
-            else                                                  -> {
+        } else {
+            _viewEvents.post(OnboardingViewEvents.OnLoginFlowRetrieved)
+        }
+    }
+
+    private suspend fun onHomeServerEdited(config: HomeServerConnectionConfig, serverTypeOverride: ServerType?, authResult: StartAuthenticationResult) {
+        when (awaitState().onboardingFlow) {
+            OnboardingFlow.SignUp -> internalRegisterAction(RegisterAction.StartRegistration) {
+                updateServerSelection(config, serverTypeOverride, authResult)
+                _viewEvents.post(OnboardingViewEvents.OnHomeserverEdited)
+            }
+            OnboardingFlow.SignIn -> {
                 updateServerSelection(config, serverTypeOverride, authResult)
-                _viewEvents.post(OnboardingViewEvents.OnLoginFlowRetrieved)
```

Comment:
this case was a legacy flow which has been removed by providing stronger typing to onAuthenticationStartedSuccess( trigger: OnboardingAction.HomeServerChange)` instead of the base `OnboardingAction`